instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubsubSingletons.java

```diff
@@ -98,4 +104,38 @@ public static void buildAndFinishSpan(Context context, PubsubMessage pubsubMessa
     Context current = subscriberInstrumenter.start(newContext, pubsubMessage);
     subscriberInstrumenter.end(current, pubsubMessage, null, null);
   }
+
+  public static Optional<Object> extractPubsubMessageAttributes(PubsubMessage pubsubMessage) {
+
+    Object attributesObject = extractAttributeFromObject(pubsubMessage, ATTRIBUTES_FIELD_NAME);
+    if (attributesObject != null) {
+      Object mapDataObject = extractAttributeFromObject(attributesObject, MAP_DATA_FIELD_NAME);
+      if (mapDataObject != null) {
+        Object delegate = extractAttributeFromObject(mapDataObject, DELEGATE_DATA_FIELD_NAME);
+        if (delegate != null) {
+          return (Optional<Object>) delegate;
+        }
+      }
+    }
+
+    return Optional.empty();
+  }
+
+  private static Object extractAttributeFromObject(Object object, String fieldName) {
+    try {
+      Class cls = object.getClass();
+      Field field = cls.getDeclaredField(fieldName);
+      field.setAccessible(true);
+      return field.get(object);
+    } catch (Exception e) {
+      System.out.println(
```

Comment:
Isn't there a logger for OTEL that can be used?

Reply:
I didn't see something similar in the other instrumentations